### PR TITLE
LPS-162179, LPS-162180, LPS-163066 | Master

### PR DESCRIPTION
**Pull request by andreefarias**
https://issues.liferay.com/browse/LPS-162179
https://issues.liferay.com/browse/LPS-162180
https://issues.liferay.com/browse/LPS-163066

### CODE REVIEW

**Comment by andreefarias** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/Object.testcase [hunk 1]
```diff
@@ -323,6 +323,63 @@ definition {
...
+		task ("Restart portal") {
+			Portlet.shutdownServer();
+
+			Portlet.startServer(deleteLiferayHome = "false");
+		}
+
+		task ("Go to Custom Object entries") {
+			Navigator.openURL();
+
+			ApplicationsMenu.gotoPortlet(
```
Comment (evanilson-santana) [inline modules/apps/object/object-test/src/testFunctional/tests/Object.testcase:372]: Hi @andreefarias you're in the correct way but you forget to assert that the custom object is present on the panel under Control Panel > User like says step 6:  `Open the global menu and **assert the custom object is present on the panel under Control Panel** > User`

File: modules/apps/object/object-test/src/testFunctional/tests/Object.testcase [hunk 2]
```diff
@@ -323,6 +323,63 @@ definition {
...
+				value = "Entry Test");
+		}
+
+		task ("Restart portal") {
+			Portlet.shutdownServer();
+
+			Portlet.startServer(deleteLiferayHome = "false");
+		}
+
+		task ("Go to Custom Object entries") {
```
Comment (evanilson-santana) [inline modules/apps/object/object-test/src/testFunctional/tests/Object.testcase:369]: Hi @andreefarias you're in the correct way but you forget to assert that the custom object is present on the panel under Control Panel > User, like says step 6:  `Open the global menu and **assert the custom object is present on the panel under Control Panel** > User`

File: modules/apps/object/object-test/src/testFunctional/tests/Object.testcase [hunk 3]
```diff
@@ -323,6 +323,63 @@ definition {
...
+			Navigator.openURL();
+
+			ApplicationsMenu.gotoPortlet(
+				category = "Users",
+				panel = "Control Panel",
+				portlet = "Custom Objects 163066");
+		}
+
+		task ("Assert that entry is present") {
```
Comment (evanilson-santana) [inline modules/apps/object/object-test/src/testFunctional/tests/Object.testcase:378]: You forget too that before asserting that a custom entry is present you need to assert that the custom object is present on the objects page like says step 7: Go to Objects and assert the custom object is present on the page

**Review by evanilson-santana — CHANGES_REQUESTED** [review]
Hi @andreefarias great job with the tests, see my comments and make the changes necessary, let me know if you have any questions.

File: modules/apps/object/object-test/src/testFunctional/tests/Object.testcase (hunk 2 above)
Comment (andreefarias) [reply modules/apps/object/object-test/src/testFunctional/tests/Object.testcase]: Fixed it. Thanks!

File: modules/apps/object/object-test/src/testFunctional/tests/Object.testcase (hunk 3 above)
Comment (andreefarias) [reply modules/apps/object/object-test/src/testFunctional/tests/Object.testcase:392]: Fixed it. Thanks!

**Comment by andreefarias** [comment]
ci:test:sf

**Comment by andreefarias** [comment]
ci:test:sf

**Review by evanilson-santana — APPROVED** [review]

File: modules/apps/object/object-test/src/testFunctional/tests/Object.testcase [hunk 4]
```diff
@@ -323,6 +323,77 @@ definition {
...
+				value = "Entry Test");
+		}
+
+		task ("Restart portal") {
+			Portlet.shutdownServer();
+
+			Portlet.startServer(deleteLiferayHome = "false");
+		}
+
+		task ("Assert that the Custom Object is present on Object admin") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/Object.testcase:369]: "Assert the Custom Object is present in Object Admin"

File: modules/apps/object/object-test/src/testFunctional/tests/Object.testcase [hunk 5]
```diff
@@ -323,6 +323,77 @@ definition {
...
+			ObjectAdmin.openObjectAdmin();
+
+			CreateObject.assertCustomObject(labelName = "Custom Object 163066");
+		}
+
+		task ("Assert that Custom Objects is present on menu portlet an go to it") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/Object.testcase:375]: "Assert the Custom Object is present on the application menu and go to it"

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 6]
```diff
@@ -1201,6 +1201,155 @@ definition {
...
+		}
+
+		task ("Create an entry via API") {
+			ObjectAdmin.addObjectSingleFieldEntryViaAPI(
+				fieldName = "customField",
+				objectName = "CustomObject162180",
+				value = "Entry Test");
+		}
+
+		task ("Assert that it's possible to see entry with ERC") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1257]: "Assert the entry is visible with an ERC column"

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 7]
```diff
@@ -1201,6 +1201,155 @@ definition {
...
+				fieldLabelName = "Custom Integer",
+				fieldName = "customInteger",
+				fieldType = "Integer",
+				isRequired = "false",
+				objectName = "CustomObject162179");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject162179");
+		}
+
+		task ("On Custom Object, go to validations tab") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1286]: "Validations "

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 8]
```diff
@@ -1201,6 +1201,155 @@ definition {
...
+			ObjectCustomViews.markViewAsDefault();
+
+			ObjectCustomViews.goToViewsBuilderTab();
+
+			ObjectCustomViews.addColumnsViaUI(addColumns = "Custom Integer,External Reference Code");
+
+			Button.clickSave();
+		}
+
+		task ("Assert that Custom Validation with 'is Greater Than' is working") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1330]: "Assert the entry is created only when the ERC value is greater than the integer value"

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 9]
```diff
@@ -1201,6 +1201,155 @@ definition {
...
+
+			Navigator.gotoBack();
+
+			ObjectPortlet.addCustomFieldEntryViaUI(
+				customField = "Custom Integer",
+				entry = "102030");
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject162179");
+
+			ObjectPortlet.viewEntry(entry = "102030");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1349]: @andreefarias what is the maximum possible value of the ERC? The integer value you use here should be greater than this.

**Review by timpak — CHANGES_REQUESTED** [review]
@andreefarias good job with these tests, can you take a look at my comments below? I changed the wording/grammar for a few of the tasks and I have a question on the ERC limit.

**Comment by andreefarias** [comment]
ci:test:sf

**Review by timpak — APPROVED** [review]
LGTM!

**Pull request closed without merging**